lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb

```diff
@@ -0,0 +1,132 @@
+module StructuraidCore
+  module DesignCodes
+    module ACI31819
+      module RC
+        module Footings
+          class PunchingCriticalSectionPerimeter
+            include DesignCodes::Utils::CodeRequirement
+            use_schema DesignCodes::Schemas::RC::Footings::PunchingCriticalSectionPerimeterSchema
+
+            def call
+              add_column_to_local_coordinates_system
+              add_perimeter_vertices_to_local_coordinates_system
+              edges_indexes = build_edges_indexes
```

Comment:
![permeter_verties_relative-to-column](https://user-images.githubusercontent.com/26731448/230082121-9ad4e602-9ccc-4ca8-9c1e-b25f9fc718f0.gif)